src/validator.ts

```diff
@@ -587,7 +587,9 @@ export class RegExpValidator {
         uFlag = false,
     ): void {
         this._uFlag = uFlag && this.ecmaVersion >= 2015
-        this._nFlag = uFlag && this.ecmaVersion >= 2018
+        this._nFlag =
+            (this._options.strict || uFlag) && this.ecmaVersion >= 2018
```

Comment:
Does https://github.com/tc39/ecma262/pull/2436 really apply all the way down to ES2018? From what I can see, the fix has only landed in ES2023 (or rather its draft). [ES2022 does not have it](https://tc39.es/ecma262/2022/multipage/text-processing.html#sec-parsepattern). I'm unfamiliar with TC39's process, so I don't know whether old(er) specs get updated at all.

However, assuming that the specs for ES2022 and before are final, I think the fix here should be this:
```suggestion
            uFlag && this.ecmaVersion >= 2018 || this._options.strict && this.ecmaVersion >= 2023
```
Otherwise, `/(?<foo>A)\k<foo>/` parsed in strict mode with `ecmaVersion == 2018` would suddenly succeed, which would not be spec-compliant.

Reply:
Hmm. I may be wrong because he is not familiar with TC39's process, but I thought it was a specification bug. However, the PR doesn't seem to be labeled as a spec-bug. I think this PR may need to be changed to branch on whether the user specifies ES2023 or not, as you say.